### ISSUE TITLE
Brackets Freezes on Large Projects

### ISSUE THREAD

**Issue by assunluis80** [adobe/brackets#12192]
Since the version 1.5, Brackets become useless. I just can't open libraries like Laravel or Lumen on it, because it freezes indefinitely and then I have to kill the process to stop it. I was searching for a solution, and I couldn't find any solution for this problem. 

I am using Ubuntu 15.10 with Gnome 3.

Thanks.

**Comment by petetnt** [comment]
Try excluding some of the heavier folders (third party folders, temporary folders...) out with an extension such as https://github.com/zaggino/brackets-file-tree-exclude. After the largest culprits are gone, you should be able to use Brackets normally.

**Comment by directrix1** [comment]
Is there a way to still use Brackets on projects with large number of files? It sure would be nice to not have to use a totally different interface to view the source of some library I'm using. Is there a way to turn off, whatever it is doing for the whole project, or just mark some folders as vendor folders so it won't scan them?

**Comment by petetnt** [comment]
Sadly I don't think there's an easy way currently but I totally feel your pain. I have an plan outlined for vastly improved blackboxing thirdparty and other similar folders here: https://github.com/adobe/brackets/issues/11833#issuecomment-149285875 but I most likely don't have time to work on it until early March (and no-one else has chosen to champion it yet AFAIK)

There's an PR involving changing the file watcher to third party lib instead of using the native node implementation in (#12190) that might or might not improve the situation. It's worth trying through.

**Comment by assunluis80** [comment]
petetnt, I have tried the plugin you suggest, but it didn't solve the problem.

**Comment by petetnt** [comment]
@assunluis80 Can you post your Health Report here (Help -> Health Report)

**Comment by swim** [comment]
Having the same issue as OP describes. Outside of an internal file black/white list the indexing brackets does seems to be super slow.

**Comment by K-Thompson** [comment]
@assunluis80 i'm wondering if you also have problem with code hints and completion? I do (see #11570). I can't prove it, but I also feel that the issue I'm having is related to the size of the project.

**Comment by assunluis80** [comment]
@K-Thompson, Yes, I do. I am having all kind of problems with the new versions.

**Comment by abose** [comment]
Tye to add the preference `"findInFiles.nodeSearch": false,` in `debug>open preferences file` and see if the issue still exisits.
I am guessing the problem is with codehints though.

**Comment by assunluis80** [comment]
petetnt here is the help report:

{
    "uuid": "aecf56d4-299f-4db9-89ad-d52105e3b417",
    "snapshotTime": 1456309083983,
    "os": "linux",
    "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/29.0.1547.80 Safari/537.36",
    "osLanguage": "en",
    "bracketsLanguage": "en-gb",
    "bracketsVersion": "1.6.0-16680",
    "fileStats": {
        "openedFileExt": {
            "csslintrc": 1,
            "css": 10,
            "js": 134,
            "less": 18,
            "html": 23,
            "php": 16,
            "ico": 1,
            "jshintrc": 26
        },
        "workingSetFileExt": {
            "js": 33,
            "less": 8,
            "css": 2,
            "html": 6,
            "php": 5,
            "jshintrc": 4
        }
    },
    "ProjectDetails": {
        "prj1068132899": {
            "numFiles": 107,
            "cacheSize": 2177399
        },
        "prj54042242": {
            "numFiles": 6088,
            "cacheSize": 30157782
        },
        "prj94539993": {
            "numFiles": 6169,
            "cacheSize": 30611634
        }
    },
    "projectLoadTimes": ":0",
    "fileOpenTimes": ":0",
    "ModuleDepsResolved": 426,
    "AppStartupTime": 427,
    "prefNodeSearchDisabled": false,
    "prefInstantSearchDisabled": false,
    "installedExtensions": [
        {
            "name": "stripper-theme",
            "version": "0.1.1"
        },
        {
            "name": "brackets-ubuntu",
            "version": "0.9.1"
        },
        {
            "name": "lintyai",
            "version": "1.2.0"
        },
        {
            "name": "camden.csslint",
            "version": "2.0.21"
        },
        {
            "name": "brackets-file-tree-exclude",
            "version": "1.0.4"
        },
        {
            "name": "camden.jshint",
            "version": "2.2.18"
        },
        {
            "name": "brackets-emmet",
            "version": "1.2.2"
        },
        {
            "name": "brackets-beautify",
            "version": "1.2.0"
        },
        {
            "name": "brackets-code-folding",
            "version": "0.3.5"
        }
    ],
    "bracketsTheme": "stripper-theme"
}

**Comment by petetnt** [comment]
@assunluis80 Both `prj54042242` and `"prj94539993` seem to contain massive amounts of files (over 6000). Can you share what kinds of files does the project consist of?

**Comment by assunluis80** [comment]
Its the Laravel framework (https://laravel.com/).

**Comment by ivan1810** [comment]
I have the same problem on Node.js/Sails.js apps

> Can you share what kinds of files does the project consist of?
"node_modules" folder in my case

**Comment by petetnt** [comment]
@assunluis80 I haven't done PHP development but make sure you ignore folder that contains `laravel` and other third party items when using the https://github.com/zaggino/brackets-file-tree-exclude extension. If you have installed it with Composer I think the folder you should ignore `vendor/`?

For @ivan1810 that above extension should work out of the box.

**Comment by assunluis80** [comment]
Excluding the folder "vendor" on plugin is helping... But I still don't have code hints and completion. :(

**Comment by guillecro** [comment]
I will put emphasis on this. Same happened to me, after using https://github.com/zaggino/brackets-file-tree-exclude now it works but nowadays projects with large dependencies are very common..

**Comment by TamerHussien** [comment]
i tried it in other way 
i moved the poject files from its location because when i open brackets it gets freeze immediately then open brackets and install https://github.com/zaggino/brackets-file-tree-exclude from extension manager with the instructions and the open the project again from the new location it works with me
